### ISSUE TITLE
Bounding box bug in MeshShape.cpp

### ISSUE THREAD

**Issue by edjohnsuk** [dartsim/dart#734]
I think there is a bug in the latest version of `MeshShape.cpp` on the Master branch:

When the `MeshShape` constructor is called, it in turn calls `setMesh(...)`, followed by `setScale(...)`. `setScale(...)` itself calls `updateVolume()`, followed by `_updateBoundingBoxDim()`. However, `updateVolume()` requires the bounding box to have already been constructed, otherwise it will compute a volume of zero. Therefore, in `setScale(...)`, `updateVolume()`, should be called _after_ `_updateBoundingBoxDim()`, not before it.

A smaller, related issue is that both `setMesh(...)` and `setScale(...)` call `_updateBoundingBoxDim()` and `updateVolume()`. But `setMesh(...)` is only found once in the code, in the `MeshShape` constructor, so will always be followed by `setScale(...)`. Therefore, it is not necessary to call either `_updateBoundingBoxDim()` or `updateVolume()` in `setMesh(...)`, because they will both be called immediately after in `setScale(..)`. And in any case, calling `_updateBoundingBoxDim()` and `updateVolume()` in `setMesh(...)` before `setScale(...)` has been called, will not do anything useful anyway, because the scale has not yet been set. This is just an efficiency issue and does not affect program execution, whereas the first issue is more serious.

In summary:
1.  In `setScale(...)`, `updateVolume()` should be called _after_ `_updateBoundingBoxDim()`.
2. In `setMesh(...)`, calls to `_updateBoundingBoxDim()` and `updateVolume()` should be removed.

**Comment by jslee02** [comment]
Resolved by #958, #959
